src/registers/mxcsr.rs

```diff
@@ -0,0 +1,120 @@
+//! Functions to read and write MXCSR register.
+
+#[cfg(feature = "instructions")]
+pub use self::x86_64::*;
+
+use bitflags::bitflags;
+
+bitflags! {
+    /// MXCSR register.
+    #[repr(transparent)]
+    pub struct MxCsr: u32 {
+        /// Invalid operation
+        const INVALID_OPERATION = 1 << 0;
+        /// Denormal
+        const DENORMAL = 1 << 1;
+        /// Divide-by-zero
+        const DIVIDE_BY_ZERO = 1 << 2;
+        /// Overflow
+        const OVERFLOW = 1 << 3;
+        /// Underflow
+        const UNDERFLOW = 1 << 4;
+        /// Precision
+        const PRECISION = 1 << 5;
+        /// Denormals are zeros
+        const DENORMALS_ARE_ZEROS = 1 << 6;
+        /// Invalid operation mask
+        const INVALID_OPERATION_MASK = 1 << 7;
+        /// Denormal mask
+        const DENORMAL_MASK = 1 << 8;
+        /// Divide-by-zero mask
+        const DIVIDE_BY_ZERO_MASK = 1 << 9;
+        /// Overflow mask
+        const OVERFLOW_MASK = 1 << 10;
+        /// Underflow mask
+        const UNDERFLOW_MASK = 1 << 11;
+        /// Precision mask
+        const PRECISION_MASK = 1 << 12;
+        /// Toward negative infinity
+        const ROUNDING_CONTROL_NEGATIVE = 1 << 13;
+        /// Toward positive infinity
+        ///
+        /// Set the both control registers for "toward zero".
+        const ROUNDING_CONTROL_POSITIVE = 1 << 14;
+        /// Flush to zero
+        const FLUSH_TO_ZERO = 1 << 15;
+    }
+}
+
+impl Default for MxCsr {
+    /// Return the default MXCSR value at reset, as documented in Intel SDM volume 2A.
+    #[inline]
+    fn default() -> Self {
+        MxCsr::from_bits_truncate(MxCsr::INVALID_OPERATION_MASK.bits |
+                                  MxCsr::DENORMAL_MASK.bits |
+                                  MxCsr::DIVIDE_BY_ZERO_MASK.bits | 
+                                  MxCsr::OVERFLOW_MASK.bits |
+                                  MxCsr::UNDERFLOW_MASK.bits |
+                                  MxCsr::PRECISION_MASK.bits)
+    }
+}
+
+#[cfg(feature = "instructions")]
+mod x86_64 {
+    use super::*;
+    use core::arch::asm;
+
+    /// Read the value of MXCSR.
+    #[inline]
+    pub fn read() -> MxCsr {
+        let mut mxcsr: u32 = 0;
+
+        #[cfg(feature = "inline_asm")]
+        unsafe {
+            asm!("stmxcsr [{}]", in(reg) &mut mxcsr, options(nostack));
+        }
+        #[cfg(not(feature = "inline_asm"))]
+        unsafe {
+            mxcsr = crate::asm::x86_64_asm_read_rflags();
```

Comment:
That's not reading `MxCsr`, that's reading `RFlags`. You'll need to edit `src/asm/mod.rs` and `src/asm/asm.s` to add the function.

Reply:
No need to apologize :) Most of the functions follow the same pattern, not a whole lot of assembly knowledge required, so it should be easy to just replicate that here. Let me now if you have any questions.

Reply:
Uh oh! LOL. I can assure that rflags part was not intentional! How embarrassing :-/